filecoin-project/go-filecoin
# Feat/mig cbor meta 2595

Use @ZenGround0 's [chainstore metadata conversion spike](https://github.com/filecoin-project/go-filecoin/tree/spike/cbor-meta) to create a first repo migration.

A fixture tarball is included for testing purposes; the same validation that is run in test is run on a migrated repo.

Validation steps:
After conversion, the old and new chain stores are loaded, then we compare old and new head tipsets, and iterate over the chainstores and ensure that the state roots are the same at each step, as well as verifying that they are both the same length.

### WARNING
This updates the repo version to 2; after this is merged, unless you reset your chain, **go-filecoin will exit until migration is run**.

#### Reviewers please help by checking out this branch and verify against a test repo:
* I used localnet to generate a small test repo, and I made a tarball of it so I could keep checking against it.
* Build and run go-filecoin against your repo and verify that it exits saying you need to migrate.
* You will want to run `go-filecoin-migrate` with the `--verbose` option, otherwise all log output goes to  `~/.filecoin-migration-logs`
* After migration, be sure to run go-filecoin daemon built from this branch to make sure that it will run with the newly migrated repo.  
* If you change back to master, build go-filecoin and run the daemon, it ought to exit because it will see that the version is too high.
